modules/storages/app/common/storages/adapters/providers/sharepoint/services/create_managed_folders_service.rb

```diff
@@ -0,0 +1,175 @@
+                  info "#{project_storage.managed_project_folder_path} does not exist. Creating..."
+                  next create_remote_folder(project_storage.managed_project_folder_name, project_storage)
+                end
+
+                rename_project_folder(folder_map[project_storage.project_folder_id], project_storage)
+              end
+
+              Success(:folder_maintenance_done)
+            end
+
+            def hide_inactive_folders(folder_map)
+              info "Hiding folders related to inactive projects"
+
+              inactive_folder_ids(folder_map).each { |item_id| hide_folder(item_id, folder_map) }
+            end
+
+            def hide_folder(file_id, folder_map)
+              info "Hiding folder with ID #{file_id} as it does not belong to any active project"
+
+              input_data = Adapters::Input::SetPermissions.build(file_id:, user_permissions: []).value_or do |failure|
+                log_validation_error(failure, file_id:, context: "hide_folder")
+              end
+
+              set_permissions.call(auth_strategy:, input_data:).or do |error|
+                add_error(:hide_inactive_folders, error, options: { context: "hide folders", path: folder_map[item_id] })
+              end
+            end
+
+            def inactive_folder_ids(folder_map)
+              folder_map.keys - @project_storages.pluck(:project_folder_id).compact
+            end
+
+            def rename_project_folder(current_folder_name, project_storage)
+              actual_name = project_storage.managed_project_folder_name
+              return if current_folder_name == actual_name
+
+              info "#{current_folder_name} is misnamed. Renaming to #{actual_name}"
+              folder_id = project_storage.project_folder_id
+
+              Adapters::Input::RenameFile.build(location: folder_id, new_name: actual_name).bind do |input_data|
+                rename_file.call(auth_strategy:, input_data:).or do |error|
+                  add_error(
+                    :rename_project_folder, error,
+                    options: { current_path: current_folder_name, project_folder_name: actual_name, project_folder_id: folder_id }
+                  )
+                end
+              end
+            end
+
+            def create_remote_folder(folder_name, project_storage)
+              input_data = Adapters::Input::CreateFolder
+                .build(folder_name:, parent_location: drive_id)
+                .value_or { return Failure(log_validation_error(it, folder_name: folder_name, parent_location: drive_id)) }
+
+              folder_info = create_folder.call(auth_strategy:, input_data:).value_or do |error|
+                add_error(:create_folder, error, options: { folder_name:, parent_location: drive_name })
+                return Failure()
+              end
+
+              project_storage.update(project_folder_id: folder_info.id)
+            end
```

Comment:
:yellow_circle:  The whole `LastProjectFolder` code that both Nextcloud and OneDrive share is not necessary here?

Reply:
I mentioned it briefly on the PR. `LastProjectFolder` handling can lead both Sharepoint and OneDrive to enter an unrecovarable state in some very specific cases.

To avoid this, I removed the handling here until we figure out a better way to deal with it.